### PR TITLE
[FEATURE] Add support for OPENSEARCH_JAVA_HOME

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
Closes https://github.com/opensearch-project/performance-analyzer-rca/issues/127

**Describe the solution you are proposing**
Add support for OPENSEARCH_JAVA_HOME (see please https://github.com/opensearch-project/OpenSearch/issues/1872), as an alternative to JAVA_HOME (both should work just fine).

**Describe alternatives you've considered**
N/A

**Additional context**
See please https://github.com/opensearch-project/OpenSearch/issues/1872

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer-rca/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: pa_bin/performance-analyzer-agent [hunk 1]
```diff
@@ -15,14 +15,23 @@ else
...
-    fi
+      fi
+  else
+    JAVA_HOME=$OPENSEARCH_JAVA_HOME
+  fi
 else
     JAVA_HOME=$2
 fi
 
+echo "Using JAVA_HOME: $JAVA_HOME"
```
Comment (reta) [inline pa_bin/performance-analyzer-agent:32]: See please https://github.com/opensearch-project/performance-analyzer/pull/133#discussion_r801062295 for the context for this change

File: docker/Dockerfile [hunk 2]
```diff
@@ -33,7 +33,7 @@ FROM centos:7 AS prep_open_search_files
 
 ENV PATH /usr/share/opensearch/bin:$PATH
 RUN curl -s https://download.java.net/java/GA/jdk11/13/GPL/openjdk-11.0.1_linux-x64_bin.tar.gz | tar -C /opt -zxf -
-ENV JAVA_HOME /opt/jdk-11.0.1
+ENV OPENSEARCH_JAVA_HOME /opt/jdk-11.0.1
```
Comment (reta) [inline docker/Dockerfile:36]: OpenSearch 1.3 already supports OPENSEARCH_JAVA_HOME, container builds and starts with no issues

File: pa_bin/performance-analyzer-agent [hunk 3]
```diff
@@ -14,19 +14,15 @@ else
...
-    JAVA_HOME=$OPENSEARCH_JAVA_HOME
-  fi
-else
-    JAVA_HOME=$2
+if [ ! -z "$2" ]; then
+  JAVA_HOME=$2
+elif [ ! -z "$OPENSEARCH_JAVA_HOME" ]; then
+  # Use OPENSEARCH_JAVA_HOME if present
+  JAVA_HOME=$OPENSEARCH_JAVA_HOME
+elif [ -z "JAVA_HOME" ]; then
```
Comment (dblock) [inline pa_bin/performance-analyzer-agent:22]: $JAVA_HOME?

**Comment by dblock** [comment]
@sruti1312 take a look pls?

**Review by dblock — APPROVED** [review]

**Review by sruti1312 — APPROVED** [review]

**Pull request merged**